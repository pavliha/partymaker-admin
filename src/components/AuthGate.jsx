import React from 'react'
import { string } from 'prop-types'
import { Redirect, Route } from 'react-router-dom'
import { connect, select } from 'src/redux'
import Storage from 'services/Storage'

const AuthGate = ({ redux: { user }, location, ...props }) => {

  Storage.put({ previous_user_location: props.path })

  if (!user) return <Redirect to="/auth/login" />

  return <Route {...props} />
}

AuthGate.propTypes = {
  path: string,
}

const redux = (state) => ({
  user: select.auth.user(state)
})

export default connect(redux)(AuthGate)
