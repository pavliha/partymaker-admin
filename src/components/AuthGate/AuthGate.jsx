import React from 'react'
import { string, elementType, shape } from 'prop-types'
import { Redirect, Route } from 'react-router-dom'
import { connect, select } from 'src/redux'
import userShape from 'shapes/user'

const AuthGate = ({ redux: { user }, ...props }) => {

  if (!user) return <Redirect to="/auth/login" />

  return <Route {...props} />
}

AuthGate.propTypes = {
  path: string,
  component: elementType,
  redux: shape({
    user: userShape,
  })
}

const redux = (state) => ({
  user: select.auth.user(state)
})

export default connect(redux)(AuthGate)
