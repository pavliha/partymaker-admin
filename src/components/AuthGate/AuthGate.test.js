import React from 'react'
import renderer from 'react-test-renderer'
import AuthGate from './AuthGate'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { object } from 'prop-types'

const loggedInState = {
  auth: { user_id: 1 },
  users: {
    entities: {
      1: {
        id: 1,
        name: 'Test User'
      }
    }
  }
}

const loggedOutState = {
  auth: { user_id: null },
  users: {
    entities: {}
  }
}

const TestGate = ({ store }) =>
  <MemoryRouter initialEntries={['/home']} initialIndex={0}>
    <Provider store={store}>
      <AuthGate path="/home" component={() => <div>route 1</div>} />
    </Provider>
  </MemoryRouter>

TestGate.propTypes = {
  store: object,
}

describe('AuthGate', () => {

  const mockStore = configureStore([])

  it('renders /home route when logged in', () => {
    const tree = renderer.create(<TestGate store={mockStore(loggedInState)} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders redirect to /auth/login when logged out', () => {
    const tree = renderer.create(<TestGate store={mockStore(loggedOutState)} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

})
