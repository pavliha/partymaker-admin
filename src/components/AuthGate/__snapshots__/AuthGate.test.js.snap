// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthGate renders /home route when logged in 1`] = `
<div>
  route 1
</div>
`;

exports[`AuthGate renders redirect to /auth/login when logged out 1`] = `null`;
